joomla/joomla-cms
# Make squeezebox initialize itself when a new subform row is added

Pull Request for Issue # .

### Summary of Changes

Subform repeatable has a function `fixScripts` which calls a few js initialization functions on a few of the field types that require it. There is a now a better way to handle that kind of thing by having any field that may need initialization listen for an event fired by subform repeatable. In this PR, I am transferring responsibility of one such initialization from subform repeatable to the correct field type itself. 

Note: this PR doesn't really fix anything. Fields that use squeezebox already work with repeatable subforms. This just does it in a somewhat better way. 

Another note: Squeezebox uses mootools and is deprecated. Bootstrap modals are prefered. Still, as long as some fields use it, they should use it like this. 

### Testing Instructions

Set up a subform containing a field that uses squeezebox (for example `contenthistory`). There's a subform in the redirect plugin that's simple to use for this purpose. 

Open up the form and create new rows. The field in each new row should function properly. 

### Expected result

Fields that use squeezebox function normally.

### Actual result

Fields that use squeezebox function normally.

### Documentation Changes Required

None
